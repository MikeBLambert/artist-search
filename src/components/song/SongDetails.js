import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const SongDetails = ({ artist, song, lyrics }) => {
  return (
    <Fragment>
      <h1>{song} by {artist}</h1>
      <p>{lyrics}</p>
    </Fragment>
  );
};

SongDetails.propTypes = {
  lyrics: PropTypes.string.isRequired
};

export default SongDetails;
