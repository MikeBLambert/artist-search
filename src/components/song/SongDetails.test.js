import React from 'react';
import { shallow } from 'enzyme';
import SongDetails from './SongDetails';

describe('SongDetails component', () => {
  it('renders a SongDetails component', () => {
    const wrapper = shallow(<SongDetails artist='The Beatles' song='Hey Jude' lyrics='Lalalalalala'/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('shows has an artist, song, and lyrics rendered', () => {
    const wrapper = shallow(<SongDetails artist='The Beatles' song='Hey Jude' lyrics='Lalalalalala'/>);
    expect(wrapper.html()).toContain('The Beatles');
  });
});
