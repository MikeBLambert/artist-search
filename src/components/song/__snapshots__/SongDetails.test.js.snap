// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongDetails component renders a SongDetails component 1`] = `
<Fragment>
  <h1>
    Hey Jude
     by 
    The Beatles
  </h1>
  <p>
    Lalalalalala
  </p>
</Fragment>
`;
